src/Features/SupportFileUploads/TemporaryUploadedFile.php

```diff
@@ -86,7 +86,7 @@ public function temporaryUrl()
         if ((FileUploadConfiguration::isUsingS3() or FileUploadConfiguration::isUsingGCS()) && ! app()->runningUnitTests()) {
             return $this->storage->temporaryUrl(
                 $this->path,
-                now()->addDay(),
+                now()->addDay()->endOfHour(),
                 ['ResponseContentDisposition' => 'filename="' . $this->getClientOriginalName() . '"']
```

Comment:
I don't understand the need for this, it's already adding a day, why do you need a few more minutes?

Reply:
@edalzell `now()->addDay()` generates a different timestamp now than it does in a minute. With `->endOfHour()`, the timestamp generated doesn't change between a request now and a subsequent request in a few minutes => the URL stays the same => the browser doesn't re-download the file. That's the whole purpose of this change, to keep the URLs stable with different request times :)

Reply:
sure but what happens if this time period spans a day? like from 11:59pm to 12:01am?

Reply:
@edalzell Then it's a new URL, but the point of this PR is to make this happen only once an hour, instead of at every new request. In #5036 I described back in the day the real-world example where this fixed all our problems (multi-file upload, preview of already uploaded files)